client/src/PublishTenderForm.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+const PublishTenderForm = () => {
+	const [title, setTitle] = useState("");
+	const [description, setDescription] = useState("");
+	const [closingDate, setClosingDate] = useState("");
+	const [announcementDate, setAnnouncementDate] = useState("");
+	const [deadlineDate, setDeadlineDate] = useState("");
+	const [skills, setSkills] = useState([]);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+
+		if (title.length < 10 || title.length > 50) {
+			alert("Tender Title must be between 10 and 50 characters.");
+			return;
+		}
+
+		if (description.length < 100 || description.length > 7500) {
+			alert("Tender Description must be between 100 and 7500 characters.");
+			return;
+		}
+
+		const today = new Date().toISOString().split("T")[0];
+		if (closingDate < today) {
+			alert("Tender Closing Date cannot be in the past.");
+			return;
+		}
+
+		if (announcementDate < closingDate) {
+			alert("Tender Announcement Date cannot be before the Closing Date.");
```

Comment:
What is the announcement date? Is it the date when they make they announce who won the tender, or the day the tender is made available?

Reply:
To my understanding, I think its the date the tender is made available due to the conditions given in the Love Me Tender Requirement Document.

Reply:
If that's the case, it should always be lower than the closing date, shouldn't it?

Reply:
Tender Announcement Date
This is a mandatory field
No max date
Min is Tender Closing Date - according to this it says that announcement date should only come after the closing. Unless I have misunderstood the statement

Reply:
we can clarify this tomorrow with the rest in our meeting and then I can make the necessary changes